src/net/sourceforge/kolmafia/textui/RuntimeLibrary.java

```diff
@@ -8205,6 +8205,14 @@ private static Value expected_damage(
         1.0 - (Math.sqrt(Math.min(1000, KoLCharacter.getDamageAbsorption()) / 10.0) - 1.0) / 10.0;
     double elementAbsorb =
         1.0 - KoLCharacter.getElementalResistance(monster.getAttackElement()) / 100.0;
+
+    // https://kol.coldfront.net/thekolwiki/index.php/Ninja_snowman_assassin
+    if (monster.getName().equals("ninja snowman assassin")) {
+      baseValue = Math.max(0, attack - defenseStat) + 120;
```

Comment:
Going by the wiki formula, Hero of the Half-Shell doesn't apply, so don't run that in the assassin case.

Reply:
good catch. I'm going to test to confirm after RO since wiki isn't explicit

Reply:
Hero of the Half-Shell applies to NSA's. Only change in tests below is having a shield equipped or not. Will update wiki

Mus 350
Mys 90
Mox 50

Expected damage WITH wicker shield: 103
Actual: 85, 77, 83, 86, 83, 71, 71, 80, 77

Expected damage with OUT shield: 189 (code currently respects hero shell for NSA)
Actual: 171, 166, 179, 166

Reply:
awesome, thanks for checking!